omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/QuickVisitFragmentController.java

```diff
@@ -56,4 +85,48 @@ public FragmentActionResult create(@SpringBean("adtService") AdtService adtServi
 		return new ObjectResult(result);
 	}
 
+	public FragmentActionResult update(@RequestParam("patientId") Patient patient,
+                       @RequestParam("visitId") Visit visit,
+                       @RequestParam(value = "selectedTypeId", required = false) VisitType selectedType,
+                       UiUtils uiUtils,
+                       HttpServletRequest request) {
+
+		BindingResult bindingResult = null;
+		if(selectedType != null) {
+			visit.setVisitType(selectedType);
+		}
+
+		if(request.getParameterNames().hasMoreElements()) {
+			List<VisitAttributeType> attributeTypes = Context.getVisitService().getAllVisitAttributeTypes();
+			WebAttributeUtil.handleSubmittedAttributesForType(visit, bindingResult, VisitAttribute.class, request, attributeTypes);
+		}
+
+		try {
+			Context.getVisitService().saveVisit(visit);
+		} catch (APIException e) {
+			request.getSession().setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "Visit.save.error");
```

Comment:
After this exception and visit save error message, won't this be overwritten by the success message on line 111?